CONTRIBUTING_CODE.md

```diff
@@ -0,0 +1,302 @@
+`probnum-master` which tracks the master branch of `upstream`.
+
+```shell
+$ git fetch upstream
+$ git checkout -b probnum-master upstream/master
+```
+
+### Code changes
+Next, create a new branch describing the feature you would like to implement. 
+```shell
+$ git checkout -b my-new-feature
+```
+You can now get started writing code in your new branch. 
+
+Commit your changes to this local branch. Make sure to keep the best practices regarding code quality in mind. 
+Next, set up a branch on your fork which is tracked by your local branch. 
+```shell
+git push --set-upstream origin my-new-feature
+```
+Once in a while, merge changes of the upstream master branch into your local `my-new-feature` branch.
+```shell
+$ git checkout probnum-master
+$ git pull
+$ git checkout my-new-feature
+$ git merge probnum-master
+```
+### Open a pull-request
+Once you are happy with your PR, make sure your branch is up-to-date with upstream `master`, and re-run all tests.
+Then, from your local branch, push the changes to your remote branch
+```shell
+$ git push
+```
+
+and do the PR via the github interface. There, briefly explain the changes as well as how they were tested. That's it!
+
+
+## Virtual environments
+
+Virtual environments (venvs) help you to get a separate, clean installation of ProbNum. 
+Each venv uses a specific Python version.
+Venvs are useful as you may need different Python versions and dependencies for different projects.
+So, first make sure that you are using the correct Python version, and also that your pip 
+installation is up to date.
+
+Install the [virtualenv](https://virtualenv.pypa.io/en/latest/) package.
+```shell
+$ pip install virtualenv
+```
+
+Go to the probnum root directory. 
+Then, create a virtual environment wit the name `venv_probnum`. 
+This uses the Python version the `python` alias is linked to. 
+```shell
+$ python -m venv venv_probnum
+```
+
+
+Activate the venv (the command below works for bash, other shells might require a different command).
+```shell
+$ source venv_probnum/bin/activate
+```
+
+Check if your venv uses the correct Python path and Python version.
+(The Python path should show something like this `/home/MyUserName/probnum/venv_probnum/bin/python`.)
+```shell
+$ which python
+$ python --version
+```
+
+
+You can deactivate the venv as well.
+```shell
+$ deactivate
+```
+
+
+
+Install probnum in your environment (activate it first). 
+```shell
+$ pip install -e .
+```
+
+If needed, install additional requirements, e.g., for testing.
+```shell
+$ pip install -e .[test]
+```
+
+Check your installation with `pip freeze`.
+
+
+It is possible to create virtual environments with most modern editors such as e.g., PyCharm. Alternatively, you can 
+add the venv you created above to your project in the editor. 
+
+## Testing
+Tests are run with the [pytest package](https://docs.pytest.org/en/stable/). 
+Make sure ProbNum is installed including test requirements in your virtual environment; 
```

Comment:
Should we explain here that this can be done with `pip install probnum[test]`?

Reply:
Yeah, I thought about that, too. But it's just the command directly above, and I did not want to be too repetitive. Not sure, I think it's understandable enough, but let me know if you think we should change it.

Reply:
How about: Make sure ProbNum is installed including test requirements (`pip install probnum[test]`) in your ...

Clarity is more important than repetition here I think. Especially for newcomers. I don't feel too strongly about this though.

Reply:
sure, I'll add it :)